cmd/internal/cli/instance_stats_linux.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2022, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cli
+
+import (
+	"errors"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/sylabs/singularity/docs"
+	"github.com/sylabs/singularity/internal/app/singularity"
+	"github.com/sylabs/singularity/pkg/cmdline"
+	"github.com/sylabs/singularity/pkg/sylog"
+)
+
+// Basic Design
+// singularity instance stats <name>
+// singularity instance stats --json <name>
+
+func init() {
+	addCmdInit(func(cmdManager *cmdline.CommandManager) {
+		cmdManager.RegisterFlagForCmd(&instanceStatsUserFlag, instanceStatsCmd)
+		cmdManager.RegisterFlagForCmd(&instanceStatsJSONFlag, instanceStatsCmd)
+	})
+}
+
+// -u|--user
+var instanceStatsUser string
+
+var instanceStatsUserFlag = cmdline.Flag{
+	ID:           "instanceStatsUserFlag",
+	Value:        &instanceStatsUser,
+	DefaultValue: "",
+	Name:         "user",
+	ShortHand:    "u",
+	Usage:        "view stats for an instance belonging to a user (root only)",
+	Tag:          "<username>",
+	EnvKeys:      []string{"USER"},
+}
+
+// -j|--json
+var instanceStatsJSON bool
+
+var instanceStatsJSONFlag = cmdline.Flag{
+	ID:           "instanceStatsJSONFlag",
+	Value:        &instanceStatsJSON,
+	DefaultValue: false,
+	Name:         "json",
+	ShortHand:    "j",
+	Usage:        "output stats in json",
+}
+
+// singularity instance stats
+var instanceStatsCmd = &cobra.Command{
+	Args:                  cobra.RangeArgs(0, 1),
+	DisableFlagsInUseLine: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// An instance name is required
+		if len(args) == 0 {
```

Comment:
For consistency, it would be better to have cobra do this check, by using `ExactArgs(1)` instead of RangeArgs, above.

The command might be extended in future to accept no args (e.g. for stats on all instances), but setting the cobra Args to the current situation keeps usage and error handling more consistent.